codegen/src/api/runtime_apis.rs

```diff
@@ -39,34 +81,53 @@ fn generate_runtime_api(
         let inputs: Vec<_> = method.inputs().enumerate().map(|(idx, input)| {
             // These are method names, which can just be '_', but struct field names can't
             // just be an underscore, so fix any such names we find to work in structs.
-            let name = if input.name == "_" {
-                format_ident!("_{}", idx)
+            let (alias_name, name) = if input.name == "_" {
```

Comment:
What about names that _start_ with a `_`, like `_foo`? These would also be perfectly valid and we want to make sure they also work ok

Reply:
(and basically, can you guarantee that no combination of valid names will lead to an issue; eg you could transform `_foo` into `s_foo` before upper_camel_casing, but what if there is already an `s_foo`? Unlikely I know, but would be good to be airtight if we can be :))